metagraph/src/graph/alignment/aligner_helper.cpp

```diff
@@ -462,29 +462,50 @@ void Alignment<NodeType>::reverse_complement(const DeBruijnGraph &graph,
 
         // if the alignment starts from a source k-mer, then traverse forwards
         // until a non-dummy k-mer is hit and check if its reverse complement exists
-        const auto *dbg_succ = dynamic_cast<const DBGSuccinct*>(&graph);
-        if (dbg_succ && rev_seq[0] == boss::BOSS::kSentinel) {
-            const auto &boss = dbg_succ->get_boss();
-            assert(rev_seq.length() == graph.get_k());
-            auto edge = dbg_succ->kmer_to_boss_index(nodes_.back());
-            auto edge_label = boss.get_W(edge) % boss.alph_size;
-
-            // TODO: for efficiency, the last outgoing edge is taken at each step.
-            // This is valid for canonical graphs since the reverse complement of
-            // the found non-dummy k-mer is guaranteed to exist, but this node may
-            // not exist in a non-canonical graph.
-            for (size_t i = 0; i < offset_; ++i) {
-                edge = boss.fwd(edge, edge_label);
-                edge_label = boss.get_W(edge) % boss.alph_size;
-                if (edge_label == boss::BOSS::kSentinelCode) {
-                    // reverse complement not found
-                    *this = Alignment();
-                    return;
+        if (rev_seq[0] == boss::BOSS::kSentinel) {
+            if (const auto *dbg_succ = dynamic_cast<const DBGSuccinct*>(&graph)) {
+                const auto &boss = dbg_succ->get_boss();
+                auto edge = dbg_succ->kmer_to_boss_index(nodes_.back());
+                auto edge_label = boss.get_W(edge) % boss.alph_size;
+
+                // TODO: for efficiency, the last outgoing edge is taken at each step.
+                // This is valid for canonical graphs since the reverse complement of
+                // the found non-dummy k-mer is guaranteed to exist, but this node may
+                // not exist in a non-canonical graph.
+                for (size_t i = 0; i < offset_; ++i) {
+                    edge = boss.fwd(edge, edge_label);
+                    edge_label = boss.get_W(edge) % boss.alph_size;
+                    if (edge_label == boss::BOSS::kSentinelCode) {
+                        // reverse complement not found
+                        *this = Alignment();
+                        return;
+                    }
+
+                    nodes_.push_back(dbg_succ->boss_to_kmer_index(edge));
+                    rev_seq.push_back(boss.decode(edge_label));
+                }
+            } else {
```

Comment:
Is `else` written for a wrapper of DBGSuccinct?

If so, add an assertion statement or throw a runtime error, or something.

Reply:
done